### DISCUSSION TITLE
SVG conversion to PNG produces a blank image

### DISCUSSION THREAD

**Discussion by kostelis** [ImageMagick/ImageMagick#7105] [Help] [answered]
### ImageMagick version

 ImageMagick 7.1.1-28 Q16-HDRI x64 365d9b6:20240211

### Operating system

Windows

### Operating system, version and so on

Microsoft Windows 11 Pro
Version 10.0.22631 Build 22631

### Description

When trying to convert the attached SVG image to PNG it results in a blank PNG image. I tried specifying the ```-define svg:xml-parse-huge=true``` flag too, but it doesn't work:

```magick C:\path\to\test.svg -define svg:xml-parse-huge=true C:\path\to\result.png```

![test](https://github.com/ImageMagick/ImageMagick/assets/29204691/37cf508c-46ea-4cf4-be0e-77e1af2e6d5c)

I found many discussion threads with blank issues when the SVG contains external images embedded into SVG, however, this file is a pure SVG file with no external references and it still fails to convert. 

Also when opening it with IMDisplay, it renders a blank image too (IMDisplay Version 1.0, ImageMagick 7.1.1-28 Q16-HDRI x64).

My ultimate goal is to convert this SVG to PNG using Magick.NET, however, I figure that the same issue happens in the CLI library as well. How could it be solved?

### Steps to Reproduce

Try to convert the attached SVG file to PNG or JPG.

### Images

Source file:

![test](https://github.com/ImageMagick/ImageMagick/assets/29204691/0bc80c23-4f98-48e3-b13a-7d9e18513c98)

Result:

![blank](https://github.com/ImageMagick/ImageMagick/assets/29204691/4fca4916-5860-43e9-9057-29acd6a36c6c)

**Comment by fmw42** [comment]
Works fine for me on IM 7.1.1.28 on Mac OSX Ventura using Inkscape in Imagemagick as the rendering delegate.

`magick test.svg -define svg:xml-parse-huge=true result.png`

![result](https://github.com/ImageMagick/ImageMagick/assets/13458218/ad296596-ac8c-4ee1-bda2-8ff0c4f1ff83)


What delegate are you using to rasterize your SVG file. IM can use any of Inkscape, RSVG, IM internal MSVG, in decreasing order of effectiveness.  If using the internal MSVG/XML delegate, try installing Inkscape. IM should use that then by default if it can find it.

**Comment by kostelis** [comment]
Hey, thanks for a quick reply! I am not using any third-party software for SVG rendering. As I mentioned, my use case is to convert the SVG to PNG using ```Magick.NET``` library for .NET C#. As I understand it is using some default internal SVG processor which does not handle the provided SVG. 

How do I make use of Inkscape in ```Magic.NET```? Do I simply need to install the default Inkscape build for Windows on my server machine, and it will use it out of the box? Or do I need some specific build for that?

**Comment by fmw42** [comment]
Just install Inkscape. It is an application, not a delegate.

**Comment by dlemstra** [comment]
The Magick.NET library uses the `rsvg` delegate by default for `svg` files. The issue with this file is that it has no background color specified and the default for ImageMagick is white. You can resolve that by using the following code:

```C#
var readSettings = new MagickReadSettings
{
    BackgroundColor = MagickColors.Transparent,
};
using var image = new MagickImage(@"C:\path\to\test.svg", readSettings);
image.Write(@"C:\path\to\result.png");
```

**Comment by kostelis** [comment]
Hey @dlemstra, thanks for your suggestion! However, the solution only partially works. 

When using Magick.NET It loses the purple part from the icon and results in this:

![test](https://github.com/ImageMagick/ImageMagick/assets/29204691/ae006be1-b06c-4fd1-aee6-bb9fe579419f)

Even if I install Inkscape it results in the same issue. Inkscape itself renders the SVG fine:

<img width="1918" alt="image" src="https://github.com/ImageMagick/ImageMagick/assets/29204691/c77eeb67-a112-459b-9263-4491433cd817">

IMDisplay still renders it as blank:

![image](https://github.com/ImageMagick/ImageMagick/assets/29204691/a6584021-a295-4568-8016-509c7b1190fa)


```inkscape``` command is available in the CLI:

<img width="322" alt="image" src="https://github.com/ImageMagick/ImageMagick/assets/29204691/d73a5f1e-f501-4e0f-af65-ff1e24ebaf55">


I wonder if the issue lies in ImageMagick? Do you have any suggestions on how to resolve it?

**Comment by dlemstra** [comment]
I am using the latest version of ImageMagick in my local development build of Magick.NET and I am getting the purple logo in my output image. And the current release of ImageMagick also shows the purple icon in imdisplay. Are you using the most recent release of both applications?

**Reply by kostelis** [reply]
Yes, I am using the latest ImageMagick on my machine (Windows 10):

```ImageMagick 7.1.1-28 Q16-HDRI x64```

<img width="267" alt="image" src="https://github.com/ImageMagick/ImageMagick/assets/29204691/23c3333a-cd28-4998-a6f2-9b87e3ca4a35">

And the Nuget package I am using in my project is ```v13.5.0```:
[Magick.NET-Q8-AnyCPU 13.5.0](https://www.nuget.org/packages/Magick.NET-Q8-AnyCPU)

Both result in the same issue. Do you have any ideas about what might be causing this?

**Comment by kostelis** [comment]
I am still struggling with SVG and ImageMagick. I have Inkscape installed on my machine and it renders the SVG files fine. For example:

<img width="1916" alt="image" src="https://github.com/ImageMagick/ImageMagick/assets/29204691/8ada2d91-1203-4735-90f8-302ddf17391d">

However, when opening the same SVG image with Image Magick viewer (IMDisplay) it still shows a broken image:

<img width="1439" alt="image" src="https://github.com/ImageMagick/ImageMagick/assets/29204691/3fdaed6e-204f-4684-91ab-f768bd6f52fa">

How do I "tell" ImageMagick to use Inkscape for SVG rendering instead of its' internal RSVG rendering library?

**Comment by fmw42** [comment]
Just install Inkscape. Imagemagick should then use then in preference to RSVG.

**Comment by snibgo** [comment]
After installing Inkscape on Windows, ensure that its path is on the system path, so that when you type "inkscape" at the console, the program runs.

**Comment by kostelis** [comment]
As mentioned in my previous posts, Inkscape is installed and works perfectly fine using both the GUI and the CLI:

> `inkscape` command is available in the CLI:
> 
> <img alt="image" width="322" src="https://private-user-images.githubusercontent.com/29204691/304356603-d73a5f1e-f501-4e0f-af65-ff1e24ebaf55.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1MTc1NjcsIm5iZiI6MTcwODUxNzI2NywicGF0aCI6Ii8yOTIwNDY5MS8zMDQzNTY2MDMtZDczYTVmMWUtZjUwMS00ZTBmLWFmNjUtZmYxZTI0ZWJhZjU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIxVDEyMDc0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI1MzM4ZTg5ZGI0NGIxOGRkMWVlNzA2YjQ5ZTBmYjViMTQwNDY3MjQxOTE2Yjk0YmY1MzM5MThjZTlmMTcyZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.U00q6z-YRI5GNihc_y4jVX3k_LWpF8Ixl277lbecPzA">

I can convert the SVGs to PNGs using the Inkscape command line with no issues at all:

```
 inkscape --export-type="png" "\path\to\sample.svg"
```

However, ImageMagick does not use it for rendering. I am not sure how should I describe the issue to be more clear.

**Comment by dlemstra** [comment, marked as answer]
It turns out that the behavior on Windows is different compared to other platforms. The use of Inkscape is disabled by default and requires a change to the `delegates.xml` file. This was done because of the error code that was returned when the `inkscape` executable could not be found would result in an exception. I just pushed a patch to ImageMagick to change this behavior and `inkscape` will be used on Windows in the next release when it is installed.

**Comment by kostelis** [comment]
Thank you, @dlemstra. Finally, we got to some sort of conclusion! 

Will this work for `Magic.NET` out of the box as well after the next release?

**Reply by dlemstra** [reply]
This will work for Magick.NET out of the box with the next release. You can already test it on the command line because that was released this weekend.

**Comment by ashifhusainoo7** [comment]
I am still facing the same issue with the Magick.NET version 14.2.0

**Reply by dlemstra** [reply]
Did you install Inkscape?

**Reply by ashifhusainoo7** [reply]
Nope, don't we have any other way around ?

**Reply by ashifhusainoo7** [reply]
For context: i have svg string in memorystream and i am using that to convert into a transparent png and again saving it to ms and further passing that ms to openshot for rendering it to video.

**Reply by dlemstra** [reply]
This issues is about Inkscape not working so it feels like you are running into something else.
